runtimes/peregrine/src/weights/did.rs

```diff
@@ -47,198 +47,145 @@ use sp_std::marker::PhantomData;
 /// Weights for did using the recommended hardware.
 pub struct WeightInfo<T>(PhantomData<T>);
 impl<T: frame_system::Config> did::WeightInfo for WeightInfo<T> {
-	fn create_ed25519_keys(n: u32, c: u32, ) -> Weight {
-		(155_554_000_u64)
-			// Standard Error: 40_000
-			.saturating_add((2_340_000_u64).saturating_mul(n as Weight))
-			// Standard Error: 12_000
-			.saturating_add((10_183_000_u64).saturating_mul(c as Weight))
+	fn create_ed25519_key() -> Weight {
+		3_207_750_000_u64
 			.saturating_add(T::DbWeight::get().reads(4_u64))
-			.saturating_add(T::DbWeight::get().writes(4_u64))
-			.saturating_add(T::DbWeight::get().writes((1_u64).saturating_mul(c as Weight)))
-	}
-	fn create_sr25519_keys(n: u32, c: u32, ) -> Weight {
-		(157_851_000_u64)
-			// Standard Error: 27_000
-			.saturating_add((2_477_000_u64).saturating_mul(n as Weight))
-			// Standard Error: 8_000
-			.saturating_add((10_523_000_u64).saturating_mul(c as Weight))
+			.saturating_add(T::DbWeight::get().writes(3_u64))
+	}
+	// Storage: System Account (r:2 w:2)
+	// Storage: Did DidBlacklist (r:1 w:0)
+	// Storage: Did Did (r:1 w:1)
+	fn create_sr25519_key() -> Weight {
+		3_220_307_000_u64
 			.saturating_add(T::DbWeight::get().reads(4_u64))
-			.saturating_add(T::DbWeight::get().writes(4_u64))
-			.saturating_add(T::DbWeight::get().writes((1_u64).saturating_mul(c as Weight)))
-	}
-	fn create_ecdsa_keys(n: u32, c: u32, ) -> Weight {
-		(275_184_000_u64)
-			// Standard Error: 62_000
-			.saturating_add((2_307_000_u64).saturating_mul(n as Weight))
-			// Standard Error: 19_000
-			.saturating_add((9_953_000_u64).saturating_mul(c as Weight))
+			.saturating_add(T::DbWeight::get().writes(3_u64))
+	}
+	// Storage: System Account (r:2 w:2)
+	// Storage: Did DidBlacklist (r:1 w:0)
+	// Storage: Did Did (r:1 w:1)
+	fn create_ecdsa_key() -> Weight {
+		5_057_009_000_u64
 			.saturating_add(T::DbWeight::get().reads(4_u64))
-			.saturating_add(T::DbWeight::get().writes(4_u64))
-			.saturating_add(T::DbWeight::get().writes((1_u64).saturating_mul(c as Weight)))
+			.saturating_add(T::DbWeight::get().writes(3_u64))
 	}
+	// Storage: Did DidEndpointsCount (r:1 w:1)
+	// Storage: Did Did (r:1 w:1)
+	// Storage: Did DidBlacklist (r:0 w:1)
+	// Storage: Did ServiceEndpoints (r:0 w:1)
 	fn delete(c: u32, ) -> Weight {
-		(40_970_000_u64)
-			// Standard Error: 4_000
-			.saturating_add((1_039_000_u64).saturating_mul(c as Weight))
+		1_097_385_000_u64
+			// Standard Error: 312_000
+			.saturating_add(30_895_000_u64.saturating_mul(c as Weight))
```

Comment:
Thats a pretty significant increase in the weight!

Reply:
These benchmark weights are temporary. They were executed on the Rust server and copy-pasted. I need to re-run them once the PR is approved. But good thing that you noticed.